Finished/PictureBot/Dialogs/SearchDialog.cs

```diff
@@ -16,16 +15,16 @@ namespace PictureBot.Dialogs
     public class SearchDialog : IDialog<object>
     {
         private string searchText = "";
-        
+        private static ISearchIndexClient indexClientForQueries = null;
```

Comment:
Consider a Lazy. Seriously though, I'm only pointing this out in case you don't know about Lazy - feel free to ignore me, there's nothing wrong with your method (at least if the instantiation block is idempotent, or the constructor isn't invoked by multiple threads at the same time).